JS/Lesson34/goksel/recipe-app/src/modules/user/RegistrationForm/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { useForm } from "react-hook-form";
+import { Link, unstable_HistoryRouter } from 'react-router-dom';
+import Avatar from '@mui/material/Avatar';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import { Box, Container, Grid, TextField, Typography, FormControl, InputLabel, OutlinedInput, InputAdornment, IconButton, FormControlLabel, Button, Snackbar, FormHelperText } from "@mui/material";
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import * as React from 'react';
+import Checkbox from '@mui/material/Checkbox';
+
+
+interface RegistrationFormProps {
+	open: boolean;
+	setOpen: any;
+	firstName: string;
+	lastName: string;
+	email: string;
+	password: string;
+}
+
+export const RegistrationForm = () => {
+
+	const [open, setOpen] = React.useState(false);
+
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+		reset
+	} = useForm<RegistrationFormProps>();
+
+
+	const onSubmit = (data: RegistrationFormProps) => {
+		console.log(data);
+		reset();
+		setOpen(true);
+	}
+
+	const [showPassword, setShowPassword] = React.useState(false);
```

Comment:
Move this line above. Whne structuring components, try to keep similar things combined. E.g. start with useState hooks, then add any other hooks that you may need (useForm in this case), then add hooks like useEffect (if needed), then other functions, then return statement with JSX.

Reply:
Fixed